unstructured/ingest/connector/sharepoint.py

```diff
@@ -31,6 +34,26 @@
 CONTENT_LABELS = ["CanvasContent1", "LayoutWebpartsContent1", "TimeCreated"]
 
 
+@dataclass
+class SharepointPermissionsConfig(BaseConfig):
+    application_id: str = None
+    client_credential: str = None
+    tenant: str = None
+
+    def __post_init__(self):
+        self.provided = False
+        if any([self.application_id or self.client_credential or self.tenant]):
+            if not all([self.application_id and self.client_credential and self.tenant]):
+                raise ValueError(
+                    "Please provide either none or all of the following optional values:\n"
```

Comment:
Can this check be part of converting the click options since this part of the code has no knowledge of the cli parameters? It can be factored into the `from_dict` method when this config is created such as this: [CliEmbeddingConfig from_dict()](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/cli/interfaces.py#L269)

Reply:
Addressed with https://github.com/Unstructured-IO/unstructured/pull/1628/commits/f3b70195d44175c6bc49f56134f29c6b75be1e19